spark3/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -113,14 +111,20 @@ public String name() {
     return icebergTable.toString();
   }
 
+  private Schema snapshotSchema() {
+    if (snapshotId != null && icebergTable instanceof BaseTable) {
+      return ((BaseTable) icebergTable).schemaForSnapshot(snapshotId);
+    } else if (asOfTimestamp != null && icebergTable instanceof BaseTable) {
+      return ((BaseTable) icebergTable).schemaForSnapshotAsOfTime(asOfTimestamp);
+    } else {
+      return icebergTable.schema();
+    }
```

Comment:
There are 4 copies of this logic. I think it would be good to put this in a util class, like `SnapshotUtil`. Maybe name the method `SnapshotUtil.schemaAsOf(Table, Snapshot)`